### PR TITLE
Pre templated stories API, and landscape half half template.

### PR DESCRIPTION
**Pull request by gmajoulet**
Pre templated stories API, and landscape half half template.

New grid layer positions (pre-templated UI) can be added in `amp-story-templates.css`.

I'm sending this PR so you can look at one example and see what the API looks like. Lmk if you have any better idea.

```
<amp-story-page>
  <amp-story-grid-layer position="landscape half-left">
    <amp-img src="asset.jpg"></amp-img>
  </amp-story-grid-layer>
  <amp-story-grid-layer position="landscape half-right">
    <p>
      ...
    </p>
  </amp-story-grid-layer>
</amp-story-page>
```

(Pictures are free of rights, they come from [this website](https://www.pexels.com/photo-license/))

Part of #21661

### CODE REVIEW

File: examples/visual-tests/amp-story/amp-story-landscape-templates.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        </amp-story-grid-layer>
+      </amp-story-page>
+
+      <amp-story-page id="page1">
+        <amp-story-grid-layer template="fill" position="landscape half-right">
+          <amp-img layout="fill" src="./img/cat2.jpg"></amp-img>
+        </amp-story-grid-layer>
+        <amp-story-grid-layer template="center" position="landscape half-left">
+          <p>Relentlessly pursues moth annoy owner until he gives you food say meow repeatedly until belly rubs, feels good drool that box? i can fit in that box but inspect anything brought into the house wack the mini furry mouse intently stare at the same spot. Flee in terror at cucumber discovered on floor shake treat bag, and trip owner up in kitchen i want food kitty scratches couch bad kitty kick up litter so under the bed groom forever, stretch tongue and leave it slightly out, blep.</p>
```
Comment (Enriqe) [inline examples/visual-tests/amp-story/amp-story-landscape-templates.html]: What is this 😂

File: examples/visual-tests/amp-story/amp-story-landscape-templates.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+module.exports = {
+  'shows half-half template with image on the left': async (page, name) => {
+    await page.waitFor(1600);
```
Comment (Enriqe) [inline examples/visual-tests/amp-story/amp-story-landscape-templates.js]: Do we need to use `waitFor(ms)` here? Does it flake if we just wait for a css class?

Comment (gmajoulet) [reply examples/visual-tests/amp-story/amp-story-landscape-templates.js]: I updated it, let's see if that works :)

Comment (Enriqe) [reply examples/visual-tests/amp-story/amp-story-landscape-templates.js]: For future reference, you can run the tests locally repeatedly to test for these flakes:

```
for i in {1..5}; do gulp visual-diff --nobuild --percy_project ampproject/<<your-username>> --percy_token <<your-percy-token>> --grep "<<your-test--names-regex>>"; done
```

**Comment by gmajoulet** [comment]
PTAL

File: extensions/amp-story/1.0/amp-story-templates.css [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** position="landscape-half-left" template. */
+[orientation=landscape] [position="landscape-half-left"] {
+  width: 50% !important;
```
Comment (newmuis) [inline extensions/amp-story/1.0/amp-story-templates.css]: Not 100% sure, but maybe it's worth either:

1. Leaving a comment here (and below in `landscape-half-right`) about the dependencies you're assuming (`position: absolute; top: 0; bottom: 0`)
2. Alternatively, specifying all the styles you need, even though they're duplicated

**Pull request merged**